tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -0,0 +1,481 @@
+                            if (!_typeSet.Contains(oldTypeMetadata.ElementType))
+                            {
+                                _typeSet.Add(oldTypeMetadata.ElementType);
+
+                                var oldElementType = _oldTypeDictionary[oldTypeMetadata.ElementType];
+                                var newElementType = _newTypeDictionary[newTypeMetadata.ElementType];
+
+                                CompareTypeMetadata(cmdlet, oldElementType, newElementType, issueLogger);
+                            }
+                        }
+                        // If the element type has changed, log an issue
+                        else
+                        {
+                            issueLogger.LogBreakingChangeIssue(
+                                cmdlet: cmdlet,
+                                severity: 0,
+                                problemId: ProblemIds.BreakingChangeProblemId.ChangedElementType,
+                                description: string.Format(Properties.Resources.ChangedElementTypeDescription,
+                                    oldProperty, oldTypeMetadata.ElementType, newTypeMetadata.ElementType),
+                                remediation: string.Format(Properties.Resources.ChangedElementTypeRemediation,
+                                    oldProperty, oldTypeMetadata.ElementType));
+                        }
+
+                        continue;
+                    }
+
+                    // Check if the type is a generic
+                    if (oldTypeMetadata.GenericTypeArguments.Count > 0 && newTypeMetadata.GenericTypeArguments.Count > 0)
+                    {
+                        // Check if the generic type has changed
+                        if (oldTypeMetadata.Name.Equals(newTypeMetadata.Name, StringComparison.OrdinalIgnoreCase))
+                        {
+                            // Check if the number of generic type arguments is the same
+                            if (oldTypeMetadata.GenericTypeArguments.Count == newTypeMetadata.GenericTypeArguments.Count)
+                            {
+                                // For each element in the generic type arguments list, make sure that the types
+                                // are the same
+                                for (int idx = 0; idx < oldTypeMetadata.GenericTypeArguments.Count; idx++)
+                                {
+                                    if (oldTypeMetadata.GenericTypeArguments[idx].Equals(newTypeMetadata.GenericTypeArguments[idx], StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        // If we have not previously seen this generic type argument,
+                                        // run this method on the type
+                                        if (!_typeSet.Contains(oldTypeMetadata.GenericTypeArguments[idx]))
+                                        {
```

Comment:
So, we are just comparing the type names for generic types as far as I can tell.  Just as we do with non-generic types, we should conside comparing the generic type arguments property by property.

For example, List<NewObject>  may contain no breaking changes from List<OldObject>, but this check will flag it as an issue because the type name of the generic type arguments don't match. LMK if I have misinterpreted the logic or this is not clear.